assets/src/scripts/monitoring-location/components/hydrograph/parameters.js

```diff
@@ -287,15 +264,13 @@ export const drawSelectionList = function(container, store, siteno) {
         if (parameter.waterAlert.hasWaterAlert) {
             drawWaterAlertRow(expansionContainerRow, siteno, parameter);
         }
-
-        // Add the sampling method selection list to the parameter loaded when the page is first opened if it has methods.
-        // This allows the user to use the row expansion toggle to see the methods when the row has never been physically clicked.
-        const primaryMethods = getPrimaryMethods(store.getState());
-        if (getSelectedParameterCode(store.getState()) === parameter.parameterCode && primaryMethods.length > 1) {
-            select(`#expansion-container-row-${parameter.parameterCode}`).call(drawMethodPicker, parameter.parameterCode, store);
-        }
     });
 
+    // Draw method picker. This can only appear in the selected parameter row because
+    // we only know the methods for the currently selected data. Let the code figure out
+    // whether to draw it and where to put it whenever the sorted IV Methods change.
+    selectionList.call(link(store, drawMethodPicker, getSortedIVMethods, store));
```

Comment:
This is really the key line. Rather than try to figure out when the drawMethodPicker should be rerendered, I am directly tying it the Redux state change whenever the methods/parameter code selection is changed. The method-picker no longer tries to directly link rerendering to the state to any code inside that module. It simply rerenders the entire method picker in the appropriate location (if appropriate) and handles the clicks on options by calling setSelectedIVMethod.

Reply:
A very smart  way to solve the issue!